numpy/random/generator.pyx

```diff
@@ -3857,12 +3865,17 @@ cdef class Generator:
                     buf[...] = x[j]
                     x[j] = x[i]
                     x[i] = buf
+            x = np.swapaxes(x, 0, axis)
         else:
             # Untyped path.
+            if axis != 0:
+                x = np.swapaxes(x, 0, axis)
```

Comment:
I've realized this doesn't work so well if `x` is a list.

@shoyer: Any suggestions for allowing `__array_function__` overrides here without letting a copy occur?

@kianasun: Sorry for the indecision. Perhaps instead we should revert this to:
```
if axis != 0:
    raise NotImplementedError("Axis argument is only supported on ndarray objects")
```
then we can solve this properly later.

Reply:
@eric-wieser Sure.